src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -64,11 +64,16 @@ PineconeConfig getConfig() {
      * @param dimension The number of dimensions for the index.
      * @param cloud The cloud provider for the index.
      * @param region The cloud region for the index.
+     * @param deletionProtection Enable or disable deletion protection for the index.
      * @return {@link IndexModel} representing the created serverless index.
      * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
      */
-    public IndexModel createServerlessIndex(String indexName, String metric, int dimension, String cloud,
-                                            String region) throws PineconeException {
+    public IndexModel createServerlessIndex(String indexName,
+                                            String metric,
+                                            int dimension,
+                                            String cloud,
+                                            String region,
+                                            DeletionProtection deletionProtection) throws PineconeException {
```

Comment:
Is this the only `createServerlessIndex` method? Do we want to add an overload so it doesn't break any clients?

Reply:
Yes, it is the only `createServerlessIndex()`. Sorry I'm not sure what you meant by `it doesn't break any clients`? Like what is clients?

Reply:
customers who have coded to the current `createServerlessIndex` signature

Reply:
I see. I was thinking that since this is a major release, a breaking change is fine.